### PR TITLE
Transactions should inherit data from their parents

### PR DESCRIPTION
**Pull request by cicdw**
This PR adds one line that ensures child/nested transactions inherit data from their parents, which they can then safely override.  The primary use case for such a thing is to allow for patterns such as:

```python
@flow 
def my_flow():
    with transaction() as txn:
        txn.set("global_state", value)
        task_one()
        task_two()
```

And now `task_one` and `task_two` can define commit/rollback hooks that look for `"global_state"` without having to individually set the value within their respective definitions.

### CODE REVIEW

File: src/prefect/transactions.py [hunk 1]
```diff
@@ -104,6 +104,7 @@ def __enter__(self):
             # either inherit from parent or set a default of eager
             if parent:
                 self.commit_mode = parent.commit_mode
+                self._stored_values = parent._stored_values.copy()
```
Comment (desertaxle) [inline src/prefect/transactions.py:107]: With this implementation, child transactions could modify values handled via reference, like the contents of a nested dictionary. Is that acceptable?

Comment (cicdw) [reply src/prefect/transactions.py:107]: Hmm good call - no I don't think that's acceptable; I suppose I need to do a deepcopy!

Comment (cicdw) [reply src/prefect/transactions.py]: Updated to deepcopy

**Review by desertaxle — APPROVED** [review]
LGTM!

**Pull request merged**